### PR TITLE
Added logic to format the example rate differently if a zero decimal currency is used

### LINKED ISSUE #2316: Issues with how rates display when a zero-decimal currency is set as the store currency.
I noticed two UI issues when a zero-decimal currency is set as the default store currency. In this case, I was using VND (Vietnamese Dong) for testing.

1. On the main currency list page, rates are always shown to 2dp, meaning when we are converting from a zero decimal currency, in most cases the rates will display as 0.00. I think that we should check if a zero-decimal currency is the default currency, and show these rates to more precision in these cases. For example, VND to USD is ~0.000043, (so either 6 or 8 DP would probably be a good precision to use). 

Note: Ignore the `CAD` rate in this screenshot, this is because a manual rate was set.

Screenshot 2021-06-24 at 11 06 20

2. On the edit currencies page, the automatic rate is not displaying properly in cases where the precision is extremely small.  In the screenshot below, the rate should display as 0.000053.
 
Screenshot 2021-06-24 at 11 08 38

### PR DESCRIPTION
**Pull request by dmallory42**
Fixes #2316 

#### Changes proposed in this Pull Request

- When a zero-decimal currency is used as the default store currency, show the exchange rates on the main enabled currencies differently (see example photos below). This is because the display of currency rates is currently formatted to 2DP, meaning when converting from a zero-decimal to non-zero-decimal rate, the rate would usually be displayed as 1.00 -> 0.00 (because of the high precision of the conversion rate).
- Display the rate in the same way in the add currrencies form, if the store uses a zero-decimal currency as default.

*Old display for Zero Decimal Currencies*:

Screenshot 2021-07-05 at 15 33 22

*New display for Zero Decimal Currencies*:

Screenshot 2021-07-05 at 15 31 12

#### Testing instructions

* Point your local instance of the client to your sandbox. Make sure you have the latest sandbox code by running `svn up .` in your sandbox's `/public_html` folder to get the live currency rates. 
* Delete the `wcpay_multi_currency_cached_currencies` option in order to force the client to refetch currency rates from the server.
* Set your store's default currency to `USD`. Go to `Settings -> Multi-Currency` and use the `Add Currencies` modal to add a few zero-decimal (e.g. Vietnamese Dong, Indonesian Rupiah) and a few non-zero-decimal currencies (e.g. Euros, British Pounds). Look at the enabled currencies list and verify that the rates displayed look correct, and display in the format `1.00 USD -> X.XX CAD` (for example).
* Click on the pencil icon next to a currency to go to the Edit Currency page. Verify the rate displayed is in the format ` Fetch rate automatically. Current rate: 1.00 USD = X.XX CAD (Last updated 12:00 UTC)`
* Change your store's default currency to a zero-decimal currency, e.g. Vietnamese Dong.
* Go back to the Enabled Currencies page. 
* Verify that Vietnamese Dong now displays at the top of the enabled currencies list, and `Default Currency` is displayed next to it where the conversion rate would normally display.
* Verify that the conversion rates between VND and the other enabled currencies look correct and are displayed in the format `1,000 VND -> X.XX CAD` (for example).
* Click on the pencil icon next to a currency to go to the Edit Currency page. Verify the rate displayed is in the format ` Fetch rate automatically. Current rate: 1,000 VND = X.XX CAD (Last updated 12:00 UTC)`

*Questions*

- Does the React code make sense? I am still relatively new to this so would appreciate any feedback!
- Is there a way to test this using Jest? I was considering doing something like setting up another snapshot where a zero-decimal currency was used as the default currency, and checking the format of the exchange rate displayed, but wasn't sure the best way to do this.
-------------------

- [x] Added changelog entry (or does not apply)
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

### CODE REVIEW

File: client/multi-currency/enabled-currencies-list/list-item.js [hunk 1]
```diff
@@ -12,15 +12,32 @@ import { Button, Icon } from '@wordpress/components';
 import DeleteButton from './delete-button';
 
 const EnabledCurrenciesListItem = ( {
+	// eslint-disable-next-line camelcase
```
Comment (jessepearson) [inline client/multi-currency/enabled-currencies-list/list-item.js:15]: Are these eslint rules needed here and on line 32? I removed them locally and am not seeing any issues.

**Review by jessepearson — APPROVED** [review]
LGTM, with one comment about the comments disabling eslint for a couple of lines. If those are needed to pass the linter in GH, then I'd say keep them, but `npm run lint:js` didn't mind them being removed.

File: client/multi-currency/enabled-currencies-list/list-item.js (hunk 1 above)
Comment (dmallory42) [reply client/multi-currency/enabled-currencies-list/list-item.js:15]: Good catch! I think that these lines were updated as part of the PR to introduce Psalm to the Client (https://github.com/Automattic/woocommerce-payments/pull/2337) and this change made these eslint lines unnecessary. Will remove them and update the PR! :)

Comment (dmallory42) [reply client/multi-currency/enabled-currencies-list/list-item.js]: Fixed in 1f9588f.

**Pull request merged**